### PR TITLE
indexer: Test preparation

### PR DESCRIPTION
**Pull request by mslipper**
Prepares the repo to land the indexer changes by mocing some additional utilities to `op-e2e` and moving `backoff` from `op-node` into `op-service`.

### CODE REVIEW

**Comment by mslipper** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3657** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/3657" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/3657?utm_source=stack-comment).

File: op-service/backoff/operation.go [hunk 1]
```diff
@@ -42,3 +43,35 @@ func Do(maxAttempts int, strategy Strategy, op Operation) error {
 		time.Sleep(strategy.Duration(attempt - 1))
 	}
 }
+
+func DoCtx(ctx context.Context, maxAttempts int, strategy Strategy, op Operation) error {
```
Comment (trianglesphere) [inline op-service/backoff/operation.go:47]: should `Do` just be a call to `DoCtx` with `context.Background` as the supplied parameter?

**Review by trianglesphere — APPROVED** [review]

File: op-service/backoff/operation.go (hunk 1 above)
Comment (mslipper) [reply op-service/backoff/operation.go:47]: Yeah good call, I'll update.

**Review by roninjin10 — COMMENTED** [review]
already merged but lgtm

**Pull request merged**